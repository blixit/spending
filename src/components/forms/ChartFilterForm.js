import React from 'react';
import styled from 'styled-components';

import Selector from 'ui/inputs/selector/Selector';

const Block = styled.div`
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  flex: 1 0 auto;
`;

class ChartFilterForm extends React.Component {
  state = {
    timeScale: 'Day'
  };

  temporalFilters = [
    { id: 1, option: 'All' },
    { id: 2, option: 'Year' },
    { id: 3, option: 'Semester' },
    { id: 4, option: 'Quarter' },
    { id: 5, option: 'Month' },
    { id: 6, option: 'Week' },
    { id: 7, option: 'Day' }
  ];

  onTimeScaleChanged = async (e) => {
    const timeScale = e.target.value;
    const { onRefresh, refreshQuery } = this.props;
    
    try {
      const response = await refreshQuery({
        timeScale
      });
      this.setState({ timeScale });
      onRefresh && onRefresh(response);
    } catch (e) {
      // errors.global = search.queryError || e.message;
      // this.setState({ hasError: true, errors });
    }
  }

  render() {
    const { timeScale } = this.state;

    return (
      <React.Fragment>
        <Block>
          Echelle de temps
          <Selector
            color='primary'
            style={{ width: '80%' }}
            selected={timeScale}
            values={this.temporalFilters}
            onChange={this.onTimeScaleChanged}
          />
        </Block>
      </React.Fragment>
    );
  }
};

export default ChartFilterForm;
