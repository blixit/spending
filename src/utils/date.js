/**
 * Formats a date for the backend
 * @param {Date} date the date to convert
 */
export const toBackendDate = (date) => {
  return date
    ? date.toLocaleDateString() + ' ' + date.toLocaleTimeString()
    : date;
};