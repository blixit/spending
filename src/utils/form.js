/**
* Removes the listed fields from the source
* @param {*} form The source object
* @param {*} fields The fields list
*/
export const clean = (form, fields) => {
  const result = {};
  // eslint-disable-next-line no-unused-vars
  for (const field in form) {
    if (!form.hasOwnProperty(field) || fields.includes(field)) {
      continue;
    }
    result[field] = form[field];
  }
  return result;
};
