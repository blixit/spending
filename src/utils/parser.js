export const stringify = data => {
  try {
    return JSON.stringify(data);
  } catch (error) {
    return null;
  };
}

export const parse = data => JSON.parse(data);