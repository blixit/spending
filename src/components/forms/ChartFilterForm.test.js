import { withTheme } from 'ui/theming/test.utils';

import render from 'testing/render';

import ChartFilterForm from './ChartFilterForm';

describe('[components/forms/filter-form] ChartFilterForm', () => {
  const props = {
    onRefresh: jest.fn(),
    refreshQuery: jest.fn()
  };

  it('should render properly', () => {
    const chartFilterForm = render(withTheme(ChartFilterForm, props));
    expect(chartFilterForm.toJSON()).toMatchSnapshot();
  });
});
