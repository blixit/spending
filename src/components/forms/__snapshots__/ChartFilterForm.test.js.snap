// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[components/forms/filter-form] ChartFilterForm should render properly 1`] = `
<div
  className="sc-bwzfXH iRHYSf"
>
  Echelle de temps
  <select
    className="sc-bdVaJa eJaiLQ"
    color="primary"
    onChange={[Function]}
    style={
      Object {
        "width": "80%",
      }
    }
    value="Day"
  >
    <option>
      All
    </option>
    <option>
      Year
    </option>
    <option>
      Semester
    </option>
    <option>
      Quarter
    </option>
    <option>
      Month
    </option>
    <option>
      Week
    </option>
    <option>
      Day
    </option>
  </select>
</div>
`;
